### PR TITLE
Lazily initialize device model to speed up `init()`

### PR DESCRIPTION
**Pull request by kylecarbs**
This was causing a relatively constant ~10ms of delay on Linux.

### CODE REVIEW

**Review by bradfitz — COMMENTED** [review]
Can you update the commit message to match our conventional form?

    hostinfo: cache device model to speed up init

And then do the git commit -s (DCO) thing?

**Comment by kylecarbs** [comment]
Ahh, yes! Will do.

**Comment by kylecarbs** [comment]
@bradfitz updated!

File: hostinfo/hostinfo.go [hunk 1]
```diff
@@ -173,8 +174,19 @@ var (
 	firewallMode      atomic.Value // of string
 )
 
-// SetDeviceModel sets the device model for use in Hostinfo updates.
-func SetDeviceModel(model string) { deviceModelAtomic.Store(model) }
```
Comment (bradfitz) [inline hostinfo/hostinfo.go:177]: Sorry, I just noticed this is being deleted but we use this from our iOS/macOS app.

Can you keep this part still?

Comment (kylecarbs) [reply hostinfo/hostinfo.go:177]: No biggie. Added it back!

**Pull request merged**